lib/vuln/php_easter_egg/engine.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Author: Aman Gupta , github.com/aman566
+
+import socket
+import socks
+import time
+import json
+import threading
+import string
+import random
+import sys
+import struct
+import re
+import os
+from OpenSSL import crypto
+import ssl
+from core.alert import *
+from core.targets import target_type
+from core.targets import target_to_host
+from core.load_modules import load_file_path
+from lib.socks_resolver.engine import getaddrinfo
+from core._time import now
+from core.log import __log_into_file
+import requests
+from lib.payload.wordlists import useragents
+
+
+def extra_requirements_dict():
+    return {
+        "php_easter_eggs_vuln_ports": [80,443],
+        "php_easter_eggs": ['?=PHPB8B5F2A0-3C92-11d3-A3A9-4C7B08C10000',
+                            '?=PHPE9568F34-D428-11d2-A769-00AA001ACF42',
+                            '?=PHPE9568F35-D428-11d2-A769-00AA001ACF42',
+                            '?=PHPE9568F36-D428-11d2-A769-00AA001ACF42',]
+    }
+
+
+def conn(targ, port, timeout_sec, socks_proxy):
+    try:
+        if socks_proxy is not None:
+            socks_version = socks.SOCKS5 if socks_proxy.startswith(
+                'socks5://') else socks.SOCKS4
+            socks_proxy = socks_proxy.rsplit('://')[1]
+            if '@' in socks_proxy:
+                socks_username = socks_proxy.rsplit(':')[0]
+                socks_password = socks_proxy.rsplit(':')[1].rsplit('@')[0]
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit('@')[1].rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[-1]), username=socks_username,
+                                        password=socks_password)
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+            else:
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[1]))
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo()
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sys.stdout.flush()
+        s.settimeout(timeout_sec)
+        s.connect((targ, port))
+        return s
+    except Exception:
+        return None
+
+
+def php_easter_eggs_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
```

Comment:
@aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')